utilities/include/vec3.hpp

```diff
@@ -1,212 +1,66 @@
 #ifndef VEC3_HPP
 #define VEC3_HPP
 
-
 #include <iostream>
-#include <cmath> //constexpr is compiler specific
-class vec3{
-    float x,y,z;
-    public:
-        constexpr vec3() : x {0}, y {0}, z {0} {}
-
-        //Cartesian coordinate Constructor
-        constexpr vec3(float X,float Y,float Z) : x {X}, y {Y}, z {Z} {}
+#include <cmath>
 
-        //norm function
-        #ifndef __GNUC__
+#include "vec4.hpp"
+#include "config.hpp"
 
-        float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
-
-        #endif
 
-        #ifdef __GNUC__
-        
-        constexpr float norm() const {
-            return sqrt(x * x + y * y + z * z);
-        }
+/*
+structs as_index,indices are used to generate appropriate xsimd::batch_constant 
+to be used for xsimd::swizzle function used inside cross product. 
+*/
+template <class T>
+struct as_index
+{
+    using type = xsimd::as_unsigned_integer_t<T>;
+};
+struct indices
+{
+    static constexpr unsigned get(unsigned i, unsigned n)
+    {
+        if(i==n-1) return i;
+        else return (i+1)%(n-1);
+    }
+};
 
-        #endif
 
```

Comment:
I don't think this many blank lines is required. 1 or 2 is fine, but anything more seems a bit excessive.